diff --git a/misc/spout2pw.sh b/misc/spout2pw.sh
index eec68bd..e183266 100755
--- a/misc/spout2pw.sh
+++ b/misc/spout2pw.sh
@@ -450,25 +450,39 @@ main() {
     check_pipewire
     find_gbm_backends
 
-    verb="$1"
+    WRAPPERS=()
 
-    case "$verb" in
-        umu-run|/*/umu-run)
-            setup_umu "$@"
-        ;;
-        wine|/*/wine|wine|/*/wine)
-            setup_wine "$@"
-        ;;
-        */steam-launch-wrapper|*/proton)
-            setup_steam "$@"
-        ;;
-        "")
-            usage
-            exit 1
-        ;;
-        *)
-            fatal "Unknown command: $verb. spout2pw only supports Steam/Proton and umu-run."
-    esac
+    if [ -n "$1" ]; then
+        usage
+        exit 1
+    fi
+
+    while
+        if [ -n "$1" ]; then
+            # FIXME
+            fatal "Unknown command: ${WRAPPERS[0]}. spout2pw only supports Steam/Proton and umu-run."
+        fi
+        
+        verb="$1"
+
+        case "$verb" in
+            umu-run|/*/umu-run)
+                setup_umu "$@"
+                break
+            ;;
+            wine|/*/wine|wine|/*/wine)
+                setup_wine "$@"
+                break
+            ;;
+            */steam-launch-wrapper|*/proton)
+                setup_steam "$@"
+                break
+            ;;
+            *)
+                WRAPPERS+=($verb)
+                shift
+        esac
+    do true; done
 
     log "App name: ${SPOUT2PW_APPNAME:-Spout2PW}"
     log "Instance name: $SPOUT2PW_INSTANCE"
@@ -479,7 +493,7 @@ main() {
     setup_env
 
     ret=0
-    "$@" || ret="$?"
+    "${WRAPPERS[@]}" "$@" || ret="$?"
     log "Command exit status: $ret"
     exit $ret
 }
